src/silx/io/h5py_utils.py

```diff
@@ -179,151 +269,270 @@ def _top_level_names(filename, include_only=group_has_end_time):
 safe_top_level_names = retry_in_subprocess()(_top_level_names)
 
 
-class File(h5py.File):
-    """Takes care of HDF5 file locking and SWMR mode without the need
-    to handle those explicitely.
+if HAS_LOCKING_ARGUMENT:
 
-    When using this class, you cannot open different files simultatiously
-    with different modes because the locking flag is an environment variable.
-    """
+    class File(h5py.File):
```

Comment:
Would there be a way to avoid duplicating most of the `__init__` method?

Reply:
I derived both from the same base class. Alternatively I can make only one class which skips calls to the `HDF5_USE_FILE_LOCKING` utility methods for newer hdf5 versions. Or I can factor out the utility methods in a separate class and only have an instance to it for older hdf5 versions. @t20100 What do you think?

Reply:
I now have  single `File` class and put all `HDF5_USE_FILE_LOCKING` stuff in a separate class.

Reply:
Sounds good to me!